### ISSUE TITLE
SIGSEGV: panic: runtime error: invalid memory address or nil pointer dereference in v1.15.2

### ISSUE THREAD

**Issue by mprimeaux** [jhump/protoreflect#572]
I've opened up `grpcurl` [issue 414](https://github.com/fullstorydev/grpcurl/issues/414), which was recently released with support for `protoreflect` [1.15.2](https://github.com/jhump/protoreflect/releases/tag/v1.15.2).

It appears a defect was introduced in `protoreflect` version `1.5.2` resulting in the following panic:

```shell
❯ cat mutate/challenge.action.add.json |  -plaintext -vv -d @ -format=json -use-reflection=false -proto ../internal/api/hydra_service.proto localhost:50503 api.Hydra.Mutate
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x100f714e8]

goroutine 1 [running]:
github.com/jhump/protoreflect/desc/protoparse.parseToProtoRecursive({0x101406c80, 0x14000195308}, {0x140001bf8e0, 0x1c}, 0x1400025f608?, 0x100f08bf4?, 0x0?)
	github.com/jhump/protoreflect@v1.15.2/desc/protoparse/parser.go:389 +0x148
github.com/jhump/protoreflect/desc/protoparse.parseToProtoRecursive.func1(0x14000365380, 0x140003b2eb0, 0x140001e3810, {0x101406c80, 0x14000195308}, 0x1013ba560?, 0x140001d9f01?)
	github.com/jhump/protoreflect@v1.15.2/desc/protoparse/parser.go:401 +0x164
github.com/jhump/protoreflect/desc/protoparse.parseToProtoRecursive({0x101406c80, 0x14000195308}, {0x16f76a75a, 0x23}, 0x14000361400?, 0x0?, 0x1400025f7b8?)
	github.com/jhump/protoreflect@v1.15.2/desc/protoparse/parser.go:402 +0x1d8
github.com/jhump/protoreflect/desc/protoparse.parseToProtosRecursive({0x101406c80, 0x14000195308}, {0x140002fed70, 0x1, 0x0?}, 0x0?, 0x0?)
	github.com/jhump/protoreflect@v1.15.2/desc/protoparse/parser.go:365 +0x80
github.com/jhump/protoreflect/desc/protoparse.Parser.ParseFiles({{0x0, 0x0, 0x0}, 0x1, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, ...}, ...)
	github.com/jhump/protoreflect@v1.15.2/desc/protoparse/parser.go:153 +0x210
github.com/fullstorydev/grpcurl.DescriptorSourceFromProtoFiles({0x0, 0x0, 0x0}, {0x140002fed70?, 0x1400025fb18?, 0x1006a3018?})
	github.com/fullstorydev/grpcurl/desc_source.go:71 +0xbc
main.main()
	github.com/fullstorydev/grpcurl/cmd/grpcurl/grpcurl.go:501 +0xca8
```

The previous version of `protoreflect` processed the same `.proto` file successfully.

```shell
❯ cat mutate/challenge.action.add.json | ~/Downloads/grpcurl -plaintext -vv -d @ -format=json -use-reflection=false -proto ../internal/api/hydra_service.proto localhost:50503 api.Hydra.Mutate

Resolved method descriptor:
// Mutates one or more underlying data structures.
rpc Mutate ( .api.MutateRequest ) returns ( .api.MutateResponse );

Request metadata to send:

Response headers received:
content-type: application/grpc

Estimated response size: 50 bytes

Response contents:
{
  "data": [
    {
          "id": "6789a357-0d07-4cf9-8025-2438d7e639f2"
        }
  ]
}

Response trailers received:
(empty)
Sent 1 request and received 1 response
```

I'll see if I can find time to debug but before I do, I wanted to get this in front of folks who know the `protoreflect` code base better than I do at this juncture.

Your help and insights are greatly appreciated.

**Comment by mprimeaux** [comment]
Looks like the panic is triggered when an import line is encountered in the `.proto` file. For example, 

```protobuf
syntax = "proto3";
import "google/protobuf/struct.proto";
```

To reproduce this, I modified the existing test named `TestBuilder_PreserveAllCommentsAfterBuild` in `builder_test.go` that was passing before with the above `import` and now it fails with the exact panic. 

The `ast2.ImportNode` [here](https://github.com/jhump/protoreflect/blob/63b3dc80de17d33462278e3692b9de3b984d8a55/desc/protoparse/parser.go#L390) assumes the import is for a local file but doesn't seem to handle the [Google protobuf types](https://github.com/protocolbuffers/protobuf/tree/main/src/google/protobuf).

I'll leave this here until @jhump or another maintainer has an opportunity to review. I still think the `nil` check [above](https://github.com/jhump/protoreflect/issues/572#issuecomment-1737716187) might be a viable option.

**Comment by srebhan** [comment]
@mprimeaux are you willing to submit a PR for the issue as you already do have the code ready!? We are hitting the same issue in Telegraf...

**Comment by mprimeaux** [comment]
@srebhan Sure. I'll submit a PR shortly.

**Comment by mprimeaux** [comment]
@jhump Sounds great and thanks for your help. Shall I just cancel my PR since you're got a handle on it?

**Comment by jhump** [comment]
Fixed in #575.
